builtin/mainmenu/content/dlg_package.lua

```diff
@@ -209,11 +211,42 @@ local function get_formspec(data)
 
 	elseif current_tab == 2 then
 		local hypertext = info.info_hypertext.head .. info.info_hypertext.body
-
 		table.insert_all(formspec, {
 			"hypertext[0,0;", W, ",", tab_body_height - 0.375,
 			";info;", core.formspec_escape(hypertext), "]",
 		})
+	elseif current_tab == 3 then
+		if not package.reviews and not data.reviews_error and not data.reviews_loading then
+			data.reviews_loading = true
+
+			contentdb.get_package_reviews(package, function(reviews)
+				if not reviews then
+					data.reviews_error = true
+				end
+				ui.update()
+			end)
+		end
+
+		if package.reviews then
+			local hypertext = package.reviews.head .. package.reviews.body
+			hypertext = hypertext:gsub("<img name=\"?blank.png\"? ",
+					"<img name=\"" .. core.hypertext_escape(defaulttexturedir) .. "blank.png\" ")
+			hypertext = hypertext:gsub("<thumbsup>",
+					"<img name=\"" .. core.hypertext_escape(defaulttexturedir) .. "contentdb_thumb_up.png\" width=24>")
+			hypertext = hypertext:gsub("<thumbsdown>",
+					"<img name=\"" .. core.hypertext_escape(defaulttexturedir) .. "contentdb_thumb_down.png\" width=24>")
+			hypertext = hypertext:gsub("<neutral>",
+					"<img name=\"" .. core.hypertext_escape(defaulttexturedir) .. "contentdb_neutral.png\" width=24>")
+			table.insert_all(formspec, {
+				"button[", W - 3.25, ",0;3,0.8;write_review;", fgettext("Leave a review..."), "]",
```

Comment:
The "Leave a review..." button shouldn't steal valuable vertical space from the text are. Perhaps you could put it to the right of the tabheader instead?